internal/reporter/slack.go

```diff
@@ -0,0 +1,143 @@
+package reporter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	commonslack "github.com/openshift/osde2e/pkg/common/slack"
+)
+
+// SlackReporter implements Reporter interface for Slack webhook notifications
+type SlackReporter struct {
+	client *commonslack.Client
+}
+
+// NewSlackReporter creates a new Slack reporter
+func NewSlackReporter() *SlackReporter {
+	return &SlackReporter{
+		client: commonslack.NewClient(),
+	}
+}
+
+// Name returns the reporter identifier
+func (s *SlackReporter) Name() string {
+	return "slack"
+}
+
+// Report sends the analysis result to Slack via webhook
+func (s *SlackReporter) Report(ctx context.Context, result *AnalysisResult, config *ReporterConfig) error {
+	if !config.Enabled {
+		return nil // Skip disabled reporters
+	}
+
+	webhookURL, ok := config.Settings["webhook_url"].(string)
+	if !ok || webhookURL == "" {
+		return fmt.Errorf("webhook_url is required and must be a string")
+	}
+
+	// Create simple message
+	message := s.formatMessage(result, config)
+
+	// Send to Slack using common package
+	if err := s.client.SendWebhook(ctx, webhookURL, message); err != nil {
+		return fmt.Errorf("failed to send to Slack: %w", err)
+	}
+
+	return nil
+}
+
+// Message represents a simple Slack message payload
+type Message struct {
+	Text string `json:"text"`
+}
+
+// formatMessage creates a simple text message for Slack
+func (s *SlackReporter) formatMessage(result *AnalysisResult, config *ReporterConfig) *Message {
+	// Create simple text message
+	statusEmoji := "✅"
+	if result.Status != "completed" || result.Error != "" {
```

Comment:
It looks like we're getting the green check emoji even when the job fails. eg https://redhat-internal.slack.com/archives/C095XQANCBD/p1761932736561999